cognite_toolkit/_cdf_tk/builders/_location.py

```diff
@@ -81,24 +79,3 @@ def build(
                 extra_sources=None,
                 warnings=warnings,
             )
-
-    def _create_file_path(self, source_path: Path, index: int, kind: str) -> Path:
-        """Helper function to prefix the filename with an index that ensures correct deployment order.
-        It also adds the kind as a suffix.
-
-        """
-        filestem = source_path.stem
-        # Get rid of the local index
-        filestem = INDEX_PATTERN.sub("", filestem)
-
-        # Increment to ensure we do not get duplicate filenames when we flatten the file
-        # structure from the module to the build directory.
-        self.resource_counter = index
```

Comment:
This was causing the bug. In the `_create_destination_path` method the `self.resource_counter` is the state. It is incremented each time this function is called. Here, it is overwritten to be the the topological order of the locations in the current modules. Thus, when two location filter filenames are the same the last will overwrite the first. In other words, here a local index is used instead of a global one.

The fix is simply to delete this method and replace it with original `_create_destination_path` as this is using a global index. Since this method is called above in the topological order, the dependencies between the filters are still kept.